### PR TITLE
Stronger regex to validate URL legitimacy for EAF Browser

### PR DESCRIPTION
**Pull request by MatthewZMD**
Rewrite
```
((and (not (string-prefix-p "/" url))
                (not (string-prefix-p "~" url))
                (string-match thing-at-point-short-url-regexp url))
           (setq app-name "browser")
           (unless (string-prefix-p "http" url)
             (setq url (concat "http://" url)))
```
and put in the funcion eaf-open-browser.

It makes sense to validate the URL legitimacy when opening EAF browser only. As mentioned above, the current interactive "F" in `eaf-open` can only open a file interactively. We don't need to worry about validating browser URL in `eaf-open`.
Replace thing-at-point-short-url-regexp with "^\\(https?:\/\/\\)?[a-z0-9]+\\([\-\.]\\{1\\}[a-z0-9]+\\)*\.[a-z]\\{2,5\\}\\(:[0-9]{1,5}\\)?\\(\/.*\\)?$", it's a better regex that validates URL legitimacy more accurately.

### CODE REVIEW

File: eaf.el [hunk 1]
```diff
@@ -537,21 +537,25 @@ We need calcuate render allocation to make sure no black border around render co
...
+  "Open an EAF application with URL, optional APP-NAME and ARGUMENTS."
   (interactive "FOpen with EAF: ")
-  ;; Try set app-name along with url if app-name is set.
-  (unless app-name
-    (cond ((string-equal url "eaf-demo")
-           (setq app-name "demo"))
-          ((string-equal url "eaf-camera")
-           (setq app-name "camera"))
-          ((string-equal url "eaf-qutebrowser")
-           (setq app-name "qutebrowser"))
```
Comment (manateelazycat) [inline eaf.el]: Why remove above modules from eaf-open?

**Comment by manateelazycat** [comment]
感谢补丁

我看了一下，大部分都没有问题，只是有一个代码块从 eaf-open 删除了

**Comment by MatthewZMD** [comment]
> 只是有一个代码块从 eaf-open 删除了

...我不小心把应该属于下一个PR的东西放到这个PR了…已修复

**Comment by manateelazycat** [comment]
当 eaf-open-browser 独立出来以后， app-name 临时变量可以去掉，我更新了一个 commit https://github.com/manateelazycat/emacs-application-framework/commit/1f80954ad99cef4665eaa602f5a1cd5a470a3775

**Comment by MatthewZMD** [comment]
Good point, 我疏忽了

**Pull request merged**